pkg/opensearch/opensearch.go

```
@@ -130,3 +149,25 @@ func (o *OSClient) SetAutoExpandIndices(vmi *vmcontrollerv1.VerrazzanoMonitoring
 
 	return ch
 }
+
+//IsOpenSearchReady returns true when all OpenSearch pods are ready, false otherwise
+func (o *OSClient) IsOpenSearchReady(vmi *vmcontrollerv1.VerrazzanoMonitoringInstance) bool {
```

we have IsGreen method, would that work instead of creating a new method?

---

The jira success criteria/solution is 
```The OpenSearch deployment should be checked for readiness before these queries occur so that no confusing error output is produced.```

When I look at [getOpenSearchClusterHealth](https://github.com/verrazzano/verrazzano-monitoring-operator/blob/master/pkg/opensearch/health.go#L124) which is being called from `IsGreen` , it is actually invoking [/_cluster/health](https://opensearch.org/docs/latest/opensearch/rest-api/cluster-health/) which seems to be more than just checking that Deployment .

Now since the functions in question (`ConfigureISM` , `MigrateOldIndices` and `SetAutoExpandIndices`), did not wait for the `IsGreen` status before, my question is 

1. Is it ok to processing in these functions to continue when the cluster status is not `Green` but pods are ready? 
2. Does it have to be `Green` or `Yellow` will also do?
3. Should the `IsOpenSearchReady` be kept as is and in `getOpenSearchClusterHealth` call this before making the api call?

---

We had initially put no 'healthy' check in these functions, because we thought: "eventually query will succeed once OS comes online, so there is no point in putting another check in front of it". 
And yes, IsGreen may be too restrictive of a check, your check is probably more in line with what we want. We may also consider squashing these error messages unless the error is actually from OpenSearch server. These are kind of background routines anyway.